com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/resource/CheckBatchLinkableResourceStorageFacade.java

```diff
@@ -26,4 +64,28 @@ public void saveResource(final StorageAwareResource resource, final IFileSystemA
       throw e;
     }
   }
+
+  @Override
+  protected boolean doesStorageExist(final StorageAwareResource resource) {
+    ResourceSet resourceSet = resource.getResourceSet();
+    URI uri = getBinaryStorageURI(resource);
+
+    return (uri != null && resourceSet.getURIConverter().exists(uri, null)) || super.doesStorageExist(resource);
+  }
+
+  @Override
+  public ResourceStorageLoadable getOrCreateResourceStorageLoadable(final StorageAwareResource resource) {
+    try {
+      ResourceSet resourceSet = resource.getResourceSet();

```

Comment:
I would rather have here
URIConverter uriConverter = resource.getResourceSet().getURIConverter()
since that is what we use later twice